datamodel/low/reference_test.go

```diff
@@ -281,10 +307,15 @@ func TestGetCircularReferenceResult_FromMappedRef(t *testing.T) {
       properties:
         nothing:
           $ref: '#/components/schemas/Nothing'
+      required:
+        - nothing
     Nothing:
       properties:
         something: 
-          $ref: '#/components/schemas/Something'`
+          $ref: '#/components/schemas/Something'
```

Comment:
All the tests in here have `required` added. Is it possible to ensure that all this code works without `required` being present? I don't want to suddenly see a significant regression where schemas that have not used `required` no longer work.

Reply:
So, the reason I added the `required` lists here is to ensure that the existing tests which detect infinite circular references retain their existing behaviour.

Without the `required` list entries, the circular references in these tests are now valid, as they can be terminated at least once during the chain, so the tests would fail without the actual test behaviour being modified.

Reply:
I've added versions of each of these tests for their equivalents without the required flags, which should ensure that both sides of the equation work. :)